<?xml version="1.0" encoding="utf-8"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="cam">
  <xacro:macro name="cam" params="connected_to:='' ns:='' rpy:='0 0 0' xyz:='0 0 0' ">
    <xacro:unless value="${connected_to == ''}">
      <joint name="${ns}_cam_joint" type="fixed">
        <parent link="${connected_to}"/>
        <child link="${ns}_cam"/>
        <origin xyz="${xyz}" rpy="${rpy}"/>
      </joint>
    </xacro:unless>

    <link name="${ns}_cam">
      <visual>
	    <geometry>
	      <mesh filename="package://pcl_filter/franka_cam_description/meshes/visual/kinect.dae"/>
	    </geometry>
      </visual>

      <!-- for simulation -->
      <inertial>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <mass value="0.0" />
        <inertia ixx="0.1" ixy="0.0" ixz="0.0" iyy="0.1" iyz="0.0" izz="0.1" />
      </inertial>
    </link>

    <link name="camera_depth_optical_frame"/>

    <joint name="{ns}_cam_depth_joint" type="fixed">
      <origin xyz="0 0 0" rpy="${-pi/2} 0 ${-pi/2}"/>
      <parent link="${ns}_cam"/>
      <child link="camera_depth_optical_frame"/>
    </joint>

  </xacro:macro>
</robot>
